app/models/manageiq/providers/ibm_terraform/configuration_manager.rb

```diff
@@ -83,13 +54,13 @@ def provider
     super || ensure_provider
   end
 
-  def ensure_provider_zone_id
-    provider.zone_id = zone_id
-  end
-
   private
 
   def ensure_provider
     build_provider.tap { |p| p.configuration_manager = self }
   end
+
+  def ensure_zone_id
+    self.zone_id = provider.zone_id unless zone_id
```

Comment:
should this just be `self.zone_id ||= provider.zone_id`

Reply:
Thanks, i have updated the code.